internal/ui/src/components/id.tsx

```diff
@@ -0,0 +1,79 @@
+"use client";
+import { Slot } from "@radix-ui/react-slot";
+import * as React from "react";
+import { cn } from "../lib/utils";
+import { useState } from "react";
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from "./ui/tooltip";
+import { CopyIcon } from "lucide-react";
+
+
+
+type copyProps = React.SVGProps<SVGSVGElement> & {
+    width?: number;
+    height?: number;
+};
+
+
+type IdProps = {
+    value?: string;
+    truncate?: number;
+    className?: string;
+};
+
+export const Id: React.FC<IdProps> = ({ className, value, truncate, ...props }) => {
+    const [isHover, setIsHover] = useState(false);
```

Comment:
we should use tailwinds `group-hover` for these, rather than css
s/o to @sean-brydon

Reply:
Did not realize this was a thing. Looking into this.